feat: display editor setting

m-ildefons opened the pull request:
![Screenshot at 2022-12-28 17-31-19](https://user-images.githubusercontent.com/17141774/209843188-ddf121b4-f238-46ba-9205-20d9fec7103c.png)


Display the configured editor, version and path using the $EDITOR environment variable similar to the way the shell version is displayed from the $SHELL variable.

fixes: #2270

## TODO

- [x] Manpage

m-ildefons committed:
feat: display editor setting

Display the configured editor, version and path using the $EDITOR
environment variable similar to the way the shell version is displayed
from the $SHELL variable.

fixes: #2270

hykilpikonna commented:
Currently, if neither `$VISUAL` nor `$EDITOR` is set, the code will return "command not found" when trying to run `$editor_full_path` when it's empty. Can you add a null check or ignore stderr?

![image](https://user-images.githubusercontent.com/22280294/209861116-8f77ea15-68aa-4415-a05a-24859f74b73c.png)


![image](https://user-images.githubusercontent.com/22280294/209860918-4d539019-e276-4c12-91c6-b3335d75d50d.png)

hykilpikonna commented:
You can add 

```bash 
[[ -z "$editor_full_path" ]] && return
```

m-ildefons commented:
Hi,
thanks for pointing this out. I corrected it and also added the options to the man page.
It will still fail to execute if `$VISUAL` (or `$EDITOR`) is set to something that isn't executable, but that would be quite a rare misconfiguration, so I'm not sure it's worth handling this case?

hykilpikonna commented:
Works!

![image](https://user-images.githubusercontent.com/22280294/209895954-b2d838f7-2174-4aad-af46-5a22f2596eb7.png)

hykilpikonna commented:
> It will still fail to execute if `$VISUAL` (or `$EDITOR`) is set to something that isn't executable, but that would be quite a rare misconfiguration, so I'm not sure it's worth handling this case?

Yea that's fine, we can assume that EDITOR is correctly configured

hykilpikonna commented:
Thank you for your contribution! 

This PR is [merged into hyfetch](https://github.com/hykilpikonna/hyfetch/commit/51ed5be1) since this repo (dylanaraps/neofetch) seems no longer maintained.

[HyFetch](https://github.com/hykilpikonna/hyfetch) is a fork of neofetch with LGBTQ pride flags, but the repo also maintains an updated version of the original neofetch, addressing many pull requests that are not merged in the original repo.

Read the ["Running Updated Original Neofetch" section](https://github.com/hykilpikonna/hyfetch#running-updated-original-neofetch) for more info!

hykilpikonna commented:
Found another bug - when the editor doesn't support `--version` option (the pico editor for example), neofetch will be stuck on this line:

![image](https://github.com/dylanaraps/neofetch/assets/22280294/f26e382b-8d2c-4831-9e82-f2989db3486b)

And by default, macOS links nano to pico:

![image](https://github.com/dylanaraps/neofetch/assets/22280294/ac920949-c6cc-43af-b6e9-b316d6cd9e7b)

Therefore, I removed the version check for editor so that editors that doesn't support `--version` will now be unstuck.
